default/python/AI/freeorion_tools/_freeorion_tools.py

```diff
@@ -349,3 +351,31 @@ def get_ship_part(part_name: str):
         warning("Could not find part %s" % part_name)
 
     return part_type
+
+
+def named_int(name: str) -> int:
+    """
+    Returns a NamedReal from FOCS.
+    If the value does not exist, reports an error and returns 1.
+    Note that we do not raise and exception so that the AI can continue, as good as it can, with outdated information.
+    This is also why we return 1, returning 0 could cause followup errors if the value is used as divisor.
+    """
+    value = fo.getNamedValue(name)
+    if not isinstance(value, int):
```

Comment:
Let's handle separately two cases:

1. Value not found (is in None?)
2. Value is found, but it is of the wrong type.

Reply:
Well, my original idea was #3879, which had two different interface functions. But since nobody wants to merge this and I was fed up with see PyCharm complain about every use of a named value, I made this simple wrappers.

Reply:
```python
value = fo.getNamedValue(name)
if value is None:
    error(f"Requested NamedInt {name}, which doesn't exist!")
elif not isinstance(value, int):
    error(f"Requested value {name} of type int got {type(value)}!")
```

Reply:
Well, if you insist on it...

Reply:
These are different errors and different reactions from user are expected for them.